File: javascript/contains-duplicate.js

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let mappy = {};
    
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] in mappy) {
            return true;
        } else {
            mappy[nums[i]] = i;
        }
    }
    return false;
};

File: python/contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        
        mappy = {}
        
        for num in nums:
            if num in mappy:
                return True
            else:
                mappy[num] = 1
        
        return False
